astropy/_erfa/erfa_generator.py

```diff
@@ -455,21 +602,35 @@ def surround(a_list, pre, post):
 
     with open(erfahfn, "r") as f:
         erfa_h = f.read()
+        print_("read erfa header")
+    if extra:
+        with open(os.path.join(templateloc or '.', extra), "r") as f:
+            erfa_h += f.read()
+        print_("read extra header")
 
     funcs = OrderedDict()
-    section_subsection_functions = re.findall(r'/\* (\w*)/(\w*) \*/\n(.*?)\n\n',
-                                              erfa_h, flags=re.DOTALL | re.MULTILINE)
+    section_subsection_functions = re.findall(
+        r'/\* (\w*)/(\w*) \*/\n(.*?)\n\n', erfa_h,
+        flags=re.DOTALL | re.MULTILINE)
     for section, subsection, functions in section_subsection_functions:
         print_("{0}.{1}".format(section, subsection))
-        if ((section == "Astronomy") or (subsection == "AngleOps")
-            or (subsection == "SphericalCartesian")
-            or (subsection == "MatrixVectorProducts")):
-            func_names = re.findall(r' (\w+)\(.*?\);', functions, flags=re.DOTALL)
+        # TODO: just compile all sections?
+        if ((section == 'Extra')
+                or (section == "Astronomy")
+                or (subsection == "AngleOps")
+                or (subsection == "SphericalCartesian")
+                or (subsection == "MatrixVectorProducts")
+                or (subsection == 'VectorOps')):
```

Comment:
The inner parentheses are not needed here

Reply:
I left them as I found them.

But talking about those: shall just take out the whole clause? We're compiling nearly everything anyway, so there is little point in terms of trying to keep it small or so.

I'll do that as a second commit, so it would be easy to revert.